modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java

```diff
@@ -1997,36 +2018,543 @@ public String setPermission(String userName, String processName, String processV
+                String eventReceiverName = dasConfigDataJOb.getString(ProcessCenterConstants.EVENT_RECEIVER_NAME);
+
+                String processAssetPath = ProcessCenterConstants.PROCESS_ASSET_ROOT + processName + "/" +
+                        processVersion;
+                Resource resource = reg.get(processAssetPath);
+                processContent = new String((byte[]) resource.getContent());
+                Document doc = stringToXML(processContent);
+
+                //  while (keys.hasNext()) {
+                // String variableName = (String) keys.next();
+                //if (Debugger.isEnabled())
+                // log.debug(variableName);
+                //String variableType = processVariablesJOb.get(variableName).toString();
+                //JSONObject processVariableJOb= (JSONObject) processVariablesArray.get(i);
+                Element rootElement = doc.getDocumentElement();
+                Element dasConfigInfoElement = append(doc, rootElement, "analytics_config_info",
+                        ProcessCenterConstants.MNS);
+                appendText(doc, dasConfigInfoElement, "processDefinitionId", ProcessCenterConstants.MNS,
+                        processDefinitionId);
+                appendText(doc, dasConfigInfoElement, "eventStreamName", ProcessCenterConstants.MNS, eventStreamName);
+                appendText(doc, dasConfigInfoElement, "eventStreamVersion", ProcessCenterConstants.MNS,
+                        eventStreamVersion);
+                appendText(doc, dasConfigInfoElement, "eventStreamDescription", ProcessCenterConstants.MNS,
+                        eventStreamDescription);
+                appendText(doc, dasConfigInfoElement, "eventStreamNickName", ProcessCenterConstants.MNS,
+                        eventStreamNickName);
+                appendText(doc, dasConfigInfoElement, "eventReceiverName", ProcessCenterConstants.MNS,
+                        eventReceiverName);
+
+                String newProcessContent = xmlToString(doc);
+                resource.setContent(newProcessContent);
+                reg.put(processAssetPath, resource);
+                // }
+                log.info("Saved das configuration details in the registry");
+                if (log.isDebugEnabled()) {
+                    log.debug("The Saved das configuration details:" + dasConfigData);
+                }
+            }
+        } catch (TransformerException | JSONException | RegistryException e) {
+            String errMsg =
+                    "Failed to save das configuration details with info,\n" + dasConfigData + "\n,to the process.rxt";
+            log.error(errMsg, e);
+            throw new ProcessCenterException(errMsg, e);
+        } catch (Exception e) {
+            String errMsg = "Failed to convert " + processContent + " registry resource to XML";
+            log.error(errMsg, e);
```

Comment:
Logging is required as the exception is here thrown to the jaggery file (where this method is called) and the exception is not logged in there. But fixed throwing the exception with errMsg. So removed  errorMsg argument in the throwing exception.